src/components/recipes/create/RecipesCreate.test.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { act } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import '@testing-library/jest-dom/extend-expect';
+import { useCreateRecipe } from 'hooks/recipes';
+import RecipesCreate from './RecipesCreate';
+import Factory from 'tests/Factory';
+import { UseMutationResult } from '@tanstack/react-query';
+import { Recipe } from 'types/resources';
+import { RequestParamsTypeCreateRecipe } from 'types/api';
+
+jest.mock('hooks/recipes', () => ({
+    useCreateRecipe: jest.fn(),
+}));
+
+const mockedUseCreateRecipe = useCreateRecipe as jest.MockedFunction<typeof useCreateRecipe>;
+
+const mockNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useNavigate: () => mockNavigate,
+}));
+
+describe('RecipesCreate Component', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        mockedUseCreateRecipe.mockReturnValue({
+            mutate: jest.fn((params, options) => {
+                if (options?.onSuccess) {
+                    options.onSuccess();
+                }
+            }),
+        } as unknown as UseMutationResult<Recipe, Error, RequestParamsTypeCreateRecipe, unknown>);
+    });
+
+    it('renders form elements correctly', () => {
+        render(
+            <BrowserRouter>
+                <RecipesCreate />
+            </BrowserRouter>,
+        );
+
+        expect(screen.getByLabelText('Name')).toBeInTheDocument();
+        expect(screen.getByLabelText('Author')).toBeInTheDocument();
+        expect(screen.getByLabelText('Ingredients')).toBeInTheDocument();
+        expect(screen.getByText('Create Recipe')).toBeInTheDocument();
+    });
+
+    it('handles form submission correctly', async () => {
+        const component = render(
+            <BrowserRouter>
+                <RecipesCreate />
+            </BrowserRouter>,
+        );
+
+        const recipe = Factory.Recipes.createRecipe();
+
+        const { container } = component;
+        const nameField = container.querySelector('input[name="name"]');
+        const authorField = container.querySelector('input[name="author_id"]');
+        const ingredientsField = container.querySelector('input[name="ingredients"]');
+        const submitButton = container.querySelector('button[type="submit"]');
+        expect(nameField).toBeInTheDocument();
+        expect(authorField).toBeInTheDocument();
+        expect(ingredientsField).toBeInTheDocument();
+        expect(submitButton).toBeInTheDocument();
+
+        await act(async () => {
+            fireEvent.change(nameField!, { target: { value: recipe.name } });
+            fireEvent.change(authorField!, { target: { value: recipe.author_id } });
+            fireEvent.change(ingredientsField!, { target: { value: recipe.ingredients.join(',') } });
+        });
+
+        await act(async () => {
+            fireEvent.click(screen.getByText('Create Recipe'));
```

Comment:
Where possible, it's best to prefer getting by role (so in this case `button`), see the testing library priority: https://testing-library.com/docs/queries/about#priority

Having looked at the FE codebase I think that's going to be trickier than hoped, but still great to aim for it.

Reply:
Also, `userEvent` is prefered over `fireEvent` as it's designed to act more like a real user: https://testing-library.com/docs/user-event/intro